spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -639,9 +639,9 @@ public <T> String predicate(UnboundPredicate<T> pred) {
         case NOT_EQ:
           return pred.ref().name() + " != " + sqlString(pred.literal());
         case STARTS_WITH:
-          return pred.ref().name() + " LIKE '" + pred.literal() + "%'";
+          return pred.ref().name() + " LIKE '" + pred.literal().value() + "%'";
```

Comment:
Without this fix, we would get double quotes for string literals. It won't be valid SQL.

Reply:
Do we have an issue with escaping single quotes within the literal here?

Reply:
I noticed an issue while comparing string representation of filters in tests.
Without this fix, it was `dep LIKE '"d1"%'`. Now it is simply `dep LIKE 'd1%'` and matches Spark.